src/NServiceBus.Transport.RabbitMQ.AcceptanceTests/When_receiving_a_message.cs

```diff
@@ -0,0 +1,66 @@
+﻿namespace NServiceBus.Transport.RabbitMQ.AcceptanceTests
+{
+    using System;
+    using System.Threading.Tasks;
+    using AcceptanceTesting;
+    using global::RabbitMQ.Client.Events;
+    using NServiceBus.AcceptanceTests;
+    using NServiceBus.AcceptanceTests.EndpointTemplates;
+    using NUnit.Framework;
+
+    class When_receiving_a_message : NServiceBusAcceptanceTest
+    {
+        [Test]
+        public async Task Should_have_access_to_BasicDeliverEventArgs()
+        {
+            var scenario = await Scenario.Define<Context>()
+                .WithEndpoint<Receiver>(b => b.When((bus, c) => bus.SendLocal(new Message())))
+                .Done(c => c.MessageReceived)
+                .Run();
+
+            Assert.True(scenario.HandlerHasAccessToBasicDeliverEventArgs, "The handler should have access to BasicDeliverEventArgs");
+        }
+    }
+
+    public class Receiver : EndpointConfigurationBuilder
+    {
+        public Receiver()
+        {
+            EndpointSetup<DefaultServer>(c => c.UseTransport<RabbitMQTransport>());
+        }
+
+        class MyEventHandler : IHandleMessages<Message>
+        {
+            public Context Context { get; set; }
+
+            public Task Handle(Message message, IMessageHandlerContext context)
+            {
+                Context.MessageReceived = true;
+
+                try
+                {
+                    context.Extensions.Get<BasicDeliverEventArgs>();
```

Comment:
Nitpick: Why not use `.TryGet` to make this a one liner?

Reply:
No reason other than it didn't occur to me to make that change after removing the extension method. Makes sense to simplify it, though.